keps/sig-multicluster/4322-cluster-inventory/README.md

```diff
@@ -0,0 +1,1010 @@
+  this area.
+* GitOps tools (ArgoCD, flux etc) are having the requirement to deploy
+  workload to multiple clusters. They either need to build the cluster
+  concept by themselves or understand cluster API from each different
+  cluster management project. A common cluster inventory API can provide
+  a thin compatible layer for different projects.
+* Operation tools or customized external consumers: this API gives a
+  common way for different clouds and vendors to define clusters,
+  providing a vendor agnostic integration point for external tools.
+* Cluster manager implementations themselves, for purposes such as
+  grouping clusters into MCS API clustersets.
+
+### Goals
+
+<!--
+List the specific goals of the KEP. What is it trying to achieve? How will we
+know that this has succeeded?
+-->
+
+* Establish a standardized cluster inventory API to represent clusters.
+* Lay the groundwork for multi-cluster tooling by providing a
+  foundational component.
+* Accommodate multiple implementations to encourage flexibility and
+  adoption.
+* Allow for future extensions and new use cases by leaving room for
+  further development and enhancements. A unified API, such as this one,
+  is most effective when platform extension authors can use it as a
+  foundational tool to create extensions compatible with multiple
+  providers.
+* Allow cluster managers of different types to share a single point inventory.
+
+
+### Non-Goals
+
+<!--
+What is out of scope for this KEP? Listing non-goals helps to focus discussion
+and make progress.
+-->
+
+* Provide a standard reference implementation.
+* Define specific implementation details beyond general API behavior.
+* Offering functionalities related to multi-cluster orchestration.
+
+
+## Proposal
+
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation. What is the desired outcome and how do we measure success?.
+The "Design Details" section below is for the real
+nitty-gritty.
+-->
+
+the API proposed by this KEP aims to
+- Provide a reliable, consistent, and automated approach for any
+  multi-cluster application (framework, toolset) to discover available
+  clusters and take actions accordingly, in a way similar to service
+  discovery works in a microservice architecture. Through the inventory,
+  the application can query for a list of clusters to access, or watch
+  for an ever-flowing stream of cluster lifecycle events which the
+  application can act upon timely, such as auto-scaling, upgrades,
+  failures, and connectivity issues.
+- Provide a simple, clear interface for human operators to understand
+  clusters under management.
+
+### Terminology
+
+- **Cluster Manager**: An entity that creates the cluster inventory API 
+  resources per member cluster, and keeps their status up-to-date. Each
```

Comment:
Do you mean "objects" rather than "resources"?  The word "resource" is often used in the kubernetes codebase to refer to a kind of object rather than an individual.

Reply:
change to object